ArduCopter/control_auto.cpp

```diff
@@ -406,8 +408,11 @@ void Copter::auto_land_run()
 
     // set motors to full range
     motors->set_desired_spool_state(AP_Motors::DESIRED_THROTTLE_UNLIMITED);
-    
-    land_run_horizontal_control();
+    if (auto_yaw_mode == AUTO_YAW_HOLD) {
+        land_run_horizontal_control();
```

Comment:
I think that it is better to explicitly describe that the values ​​differ between true and false. That's because you can clearly see the place to understand the process.